server/internal/functions/deploy_fly.go

```diff
@@ -498,6 +506,105 @@ func (f *FlyRunner) Deploy(ctx context.Context, req RunnerDeployRequest) (res *R
 	}, nil
 }
 
+func (f *FlyRunner) Reap(ctx context.Context, req ReapRequest) error {
+	logger := f.logger.With(
+		attr.SlogProjectID(req.ProjectID.String()),
+		attr.SlogDeploymentID(req.DeploymentID.String()),
+		attr.SlogDeploymentFunctionsID(req.FunctionID.String()),
+	)
+
+	appsRepo := repo.New(f.db)
+
+	if err := f.reap(ctx, logger, appsRepo, req); err != nil {
+		return oops.E(oops.CodeUnexpected, err, "failed to reap app").Log(ctx, logger)
+	}
+
+	return nil
+}
+
+func (f *FlyRunner) reap(ctx context.Context, logger *slog.Logger, appsRepo *repo.Queries, req ReapRequest) error {
+	app, err := appsRepo.GetFlyAppNameForFunction(ctx, repo.GetFlyAppNameForFunctionParams{
+		ProjectID:    req.ProjectID,
+		DeploymentID: req.DeploymentID,
+		FunctionID:   req.FunctionID,
+	})
+	switch {
+	case errors.Is(err, sql.ErrNoRows):
+		logger.InfoContext(ctx, "no existing app found")
+		return nil
+	case err != nil:
+		return fmt.Errorf("get existing app name: %w", err)
+	}
+
+	enabled, err := appsRepo.IsReapingEnabledForProject(ctx, repo.IsReapingEnabledForProjectParams{
+		ProjectID:       req.ProjectID,
+		OrganizationIds: []string{"5ad61845-b72e-4f0d-9dde-e0bdcf98e374", "5a25158b-24dc-4d49-b03d-e85acfbea59c"},
+	})
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		return fmt.Errorf("check reaping enabled for project: %w", err)
+	}
+	if !enabled {
+		logger.InfoContext(ctx, "reaping is not enabled for project")
+		return nil
+	}
```

Comment:
calling attention to this. it is a temporary measure that limits organization for which reaping will occur. limiting blast radius for now.